internal/backend/rest/rest.go

```diff
@@ -114,12 +115,11 @@ func (b *restBackend) Save(ctx context.Context, h restic.Handle, rd io.Reader) (
 	defer cancel()
 
 	// make sure that client.Post() cannot close the reader by wrapping it
-	rd = ioutil.NopCloser(rd)
-
-	req, err := http.NewRequest(http.MethodPost, b.Filename(h), rd)
+	req, err := http.NewRequest(http.MethodPost, b.Filename(h), ioutil.NopCloser(rd))
```

Comment:
Is `ioutil.NopCloser` wrapper still necessary? `RewindReader` does not implement `Closer` interface as far as I can tell, so client shouldn't be able to close the underlying file already. (same question applies to azure and s3).

Reply:
Good question! I think it isn't strictly necessary, for the reason you mentioned (`RewindReader` does not imply that there's a `Close()` method, but there's no reason there couldn't be a `Close()`). But if we, at some point, add some implementation of `RewindReader` which also has a `Close` method, retries will suddenly stop working mysteriously because the Go stdlib HTTP client will close the body reader for e.g. `POST` requests. That's why the `NopCloser` is there.

So, although it's not strictly necessary, I'd like to keep it in to prevent future bugs.